pages/api/site/reqtenant.js

```diff
@@ -4,57 +4,92 @@ const mongoose = require("mongoose");
 var Tenant = require("../../../models/tenant");
 var Sites = require("../../../models/sites");
 var History = require("../../../models/history");
+import { auth } from "../../../utility/auth";
 const jwt = require("jsonwebtoken");
-
+const {isEmail} = require("validator")
 const bcrypt = require("bcrypt");
 const config = require("../../../config/config");
 
 export default async function handler(req, res) {
-  if (req.method === "POST") {
-    var siteId = "61ec6495892136db190346f4";
-    var siteData = await Sites.findById(siteId);
-    if (siteData) {
-      if (siteData.status === "0") {
-        var tenantData = await Tenant.find({ email: req.body.email || "" });
-        if (tenantData) {
-          var newHist = new History({
-            site_id: siteId,
-            tenant_id: tenantData[0]?._id,
-            requested_at: Date.now(),
-          });
+  
+  if(req.method == "POST"){
+    var siteId = req.body.siteId
+    var email = req.body.email
+
+    if(!isEmail(email)){
+      return sendError(res,"invalid email",500)
+    }
+
+    Sites.findById(siteId, function(err,siteData){
+      if(err)return sendError(res,err.message,500)
+      else{
 
-          var histData = await newHist.save();
+        const bearerHeader = req.headers['authorization'];
+        if(typeof bearerHeader !== 'undefined') {
```

Comment:
For this I have created one Auth function just check for it or implement a middleware for this .